bolt-modules/boltlib/lib/puppet/datatypes/target.rb

```diff
@@ -34,5 +35,8 @@
     }
   PUPPET
 
+  # Needed for Puppet to recognize targets as Puppet objects when deserializing
+  Bolt::Target.include(Puppet::Pops::Types::PuppetObject)
```

Comment:
This should be `target_implementation_class.include(Puppet::Pops::Types::PuppetObject)` so it stays unattached to any particular implementation class. We can also remove the `require 'bolt/target'` above.